benchmarks/layeredflow/layeredflow.cc

```diff
@@ -0,0 +1,471 @@
+        material_model
+          = dynamic_cast<const LayeredFlowMaterial<dim> *>(&this->get_material_model());
+
+        ref_func.reset (new AnalyticSolutions::FunctionLayeredFlow<dim>(material_model->get_beta(),
+                                                                        material_model->get_epsilon()));
+      }
+
+      const QGauss<dim> quadrature_formula (this->introspection().polynomial_degree.velocities+2);
+
+      Vector<float> cellwise_errors_u (this->get_triangulation().n_active_cells());
+      Vector<float> cellwise_errors_p (this->get_triangulation().n_active_cells());
+      Vector<float> cellwise_errors_ul2 (this->get_triangulation().n_active_cells());
+      Vector<float> cellwise_errors_pl2 (this->get_triangulation().n_active_cells());
+
+      ComponentSelectFunction<dim> comp_u(std::pair<unsigned int, unsigned int>(0,dim),
+                                          dim+2);
+      ComponentSelectFunction<dim> comp_p(dim, dim+2);
+
+      VectorTools::integrate_difference (this->get_mapping(),this->get_dof_handler(),
+                                         this->get_solution(),
+                                         *ref_func,
+                                         cellwise_errors_u,
+                                         quadrature_formula,
+                                         VectorTools::L1_norm,
+                                         &comp_u);
+      VectorTools::integrate_difference (this->get_mapping(),this->get_dof_handler(),
+                                         this->get_solution(),
+                                         *ref_func,
+                                         cellwise_errors_p,
+                                         quadrature_formula,
+                                         VectorTools::L1_norm,
+                                         &comp_p);
+      VectorTools::integrate_difference (this->get_mapping(),this->get_dof_handler(),
+                                         this->get_solution(),
+                                         *ref_func,
+                                         cellwise_errors_ul2,
+                                         quadrature_formula,
+                                         VectorTools::L2_norm,
+                                         &comp_u);
+      VectorTools::integrate_difference (this->get_mapping(),this->get_dof_handler(),
+                                         this->get_solution(),
+                                         *ref_func,
+                                         cellwise_errors_pl2,
+                                         quadrature_formula,
+                                         VectorTools::L2_norm,
+                                         &comp_p);
+
+      const double u_l1 =  Utilities::MPI::sum(cellwise_errors_u.l1_norm(),this->get_mpi_communicator());
+      const double p_l1 =  Utilities::MPI::sum(cellwise_errors_p.l1_norm(),this->get_mpi_communicator());
```

Comment:
I think looking at L1 errors is not particularly useful. We did it for one of the benchmarks because the source did it. Maybe think about removing this?

Reply:
indeed. I kept it in because the burstedde benchmark computes L1 norms. removed.